pico/tools/src/main/java/io/helidon/pico/tools/DefaultActivatorCreator.java

```diff
@@ -0,0 +1,1237 @@
+                                                                TypeName serviceTypeName,
+                                                                LazyValue<ScanResult> scan) {
+        ActivatorCreatorCodeGen codeGen = req.codeGen();
+        String template = templateHelper().safeLoadTemplate(req.templateName(), SERVICE_PROVIDER_ACTIVATOR_HBS);
+        ServiceInfoBasics serviceInfo = toServiceInfo(serviceTypeName, codeGen);
+        TypeName activatorTypeName = toActivatorTypeName(serviceTypeName);
+        TypeName parentTypeName = toParentTypeName(serviceTypeName, codeGen);
+        String activatorGenericDecl = toActivatorGenericDecl(serviceTypeName, codeGen);
+        DependenciesInfo dependencies = toDependencies(serviceTypeName, codeGen);
+        DependenciesInfo parentDependencies = toDependencies(parentTypeName, codeGen);
+        Set<String> scopeTypeNames = toScopeTypeNames(serviceTypeName, codeGen);
+        String generatedSticker = toGeneratedSticker(req);
+        List<String> description = toDescription(serviceTypeName);
+        Double weightedPriority = toWeightedPriority(serviceTypeName, codeGen);
+        Integer runLevel = toRunLevel(serviceTypeName, codeGen);
+        String postConstructMethodName = toPostConstructMethodName(serviceTypeName, codeGen);
+        String preDestroyMethodName = toPreDestroyMethodName(serviceTypeName, codeGen);
+        List<?> serviceTypeInjectionOrder = toServiceTypeHierarchy(serviceTypeName, codeGen, scan);
+        List<String> extraCodeGen = toExtraCodeGen(serviceTypeName, codeGen);
+        boolean isProvider = toIsProvider(serviceTypeName, codeGen);
+        boolean isConcrete = toIsConcrete(serviceTypeName, codeGen);
+        boolean isSupportsJsr330InStrictMode = req.configOptions().isSupportsJsr330InStrictMode();
+        Collection<Object> injectionPointsSkippedInParent =
+                toCodegenInjectMethodsSkippedInParent(isSupportsJsr330InStrictMode, activatorTypeName, codeGen, scan);
+
+        ActivatorCreatorArgs args = DefaultActivatorCreatorArgs.builder()
+                .template(template)
+                .serviceTypeName(serviceTypeName)
+                .activatorTypeName(activatorTypeName)
+                .activatorGenericDecl(Optional.ofNullable(activatorGenericDecl))
+                .parentTypeName(Optional.ofNullable(parentTypeName))
+                .scopeTypeNames(scopeTypeNames)
+                .description(description)
+                .serviceInfo(serviceInfo)
+                .dependencies(Optional.ofNullable(dependencies))
+                .parentDependencies(Optional.ofNullable(parentDependencies))
+                .injectionPointsSkippedInParent(injectionPointsSkippedInParent)
+                .serviceTypeInjectionOrder(serviceTypeInjectionOrder)
+                .generatedSticker(generatedSticker)
+                .weightedPriority(Optional.ofNullable(weightedPriority))
```

Comment:
We should avoid using Optional as a parameter

Reply:
Ah this is actually already merged builder, created issue [#6401](https://github.com/helidon-io/helidon/issues/6401)

Reply:
When an optional is declared on the builder, it will overload the setter to accept both/either:

```java
		/**
		 * Setter for 'weightedPriority'.
		 *
		 * @param val the new value
		 * @return this fluent builder
		 */
		public B weightedPriority(java.util.Optional<java.lang.Double> val) {
			this.weightedPriority = Objects.requireNonNull(val);
			return identity();
		}

		/**
		 * Setter for 'weightedPriority'.
		 *
		 * @param val the new value
		 * @return this fluent builder
		 */
		public B weightedPriority(java.lang.Double val) {
			Objects.requireNonNull(val);
			return weightedPriority(java.util.Optional.of(val));
		}
```

Reply:
I don't think it should `You should almost never use it as a field of something or a method parameter.` - [Brian Getz](https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555)

Reply:
I think even if it is Optional, there should be no method with Optional type parameter.

Reply:
Think about this code for a second:

```java
        ActivatorCreatorArgs args = DefaultActivatorCreatorArgs.builder()
                .template(template)
                .serviceTypeName(serviceTypeName)
                .activatorTypeName(activatorTypeName)
                .activatorGenericDecl(Optional.ofNullable(activatorGenericDecl))
                .parentTypeName(Optional.ofNullable(parentTypeName))
                .scopeTypeNames(scopeTypeNames)
                .description(description)
                .serviceInfo(serviceInfo)
                .dependencies(Optional.ofNullable(dependencies))
                .parentDependencies(Optional.ofNullable(parentDependencies))
                .injectionPointsSkippedInParent(injectionPointsSkippedInParent)
                .serviceTypeInjectionOrder(serviceTypeInjectionOrder)
                .generatedSticker(generatedSticker)
                .weightedPriority(Optional.ofNullable(weightedPriority))
                .runLevel(Optional.ofNullable(runLevel))
                .postConstructMethodName(Optional.ofNullable(postConstructMethodName))
                .preDestroyMethodName(Optional.ofNullable(preDestroyMethodName))
                .extraCodeGen(extraCodeGen)
                .concrete(isConcrete)
                .provider(isProvider)
                .supportsJsr330InStrictMode(isSupportsJsr330InStrictMode)
                .build();
```

If optionals were not available as arguments to the setter I would need to write if-then code for literally every method that might be nullable, and that would be rather annoying and clutter up the code.  Brian is a smart guy, sure, but I don't agree with his conclusion in this case.

Reply:
I think @Verdent is just saying that [this is an unambiguous part of our development guidelines](https://github.com/helidon-io/helidon/blob/helidon-3.x/DEV-GUIDELINES.md#:~:text=never%20use%20java.util.Optional%20as%20a%20parameter%20type).

Reply:
This is more of a policy decision like `allowNulls`.

```java
    /**
     * Should the bean and the builder allow for the possibility of nullable non-{@link java.util.Optional} values to be present.
     * Default is {@code false}.
     *
     * @return true to allow for the possibility of nullable non-Optional values to be present
     */
    boolean allowNulls() default DEFAULT_ALLOW_NULLS;
```

So I suppose we can do the same for `allowOptionals`.